### ISSUE TITLE
Get Python result without assigning to variable or using print on Python side

### ISSUE THREAD

**Issue by TyGu1** [rstudio/reticulate#595]
First, i asked on #SO: https://stackoverflow.com/questions/57938046/reticulate-get-python-result-without-assigning-to-variable

If possible i would like to print the result of python Code in R (without assigning to a variable).
This works:
```
library(reticualte)
py_run_string("print(2)")
2

```

This works:
```
p = py_run_string("x = 2")
p$x
2

```

I would like this to work:

```
py_run_string("2")
2

```
Background:
I would like to read in complete python Code and capture the Output even if (print) is not used.
If i open a Python3.7 Shell and just execute "2" as a command, the Output will be "2". Here its empty.

**Comment by HashRocketSyntax** [comment]
TLDR; a way to directly `return` the python object without variable assignment

**Comment by kevinushey** [comment]
One way to accomplish this is by using `compile` + `eval` together, e.g. something like:

```

library(reticulate)

py_evaluate <- function(code) {
  builtins <- import_builtins(convert = TRUE)
  globals <- py_eval("globals()", convert = FALSE)
  locals <- globals
  parsed <- builtins$compile(code, "<string>", "single")
  builtins$eval(parsed, globals, locals)
}

py_evaluate("2")
```

That said, in my experience simulating what a Python interpreter might do is a bit challenging (there's a bunch of edge cases). You can look at the code for `repl_python()` to see how we approached this problem.
